src/engraving/dom/stafftypechange.cpp

```diff
@@ -133,9 +133,21 @@ bool StaffTypeChange::setProperty(Pid propertyId, const PropertyValue& v)
     case Pid::STEP_OFFSET:
         m_staffType->setStepOffset(v.toInt());
         break;
-    case Pid::STAFF_LINES:
-        m_staffType->setLines(v.toInt());
-        break;
+    case Pid::STAFF_LINES: {
+        int linesOld = m_staffType->lines();
+        int linesNew = v.toInt();
+        Spatium yOffset = m_staffType->yoffset();
+
+        // if change to one line staff, center by setting y offset 2
+        if (linesNew == 1 && linesOld != 1) {
+            m_staffType->setYoffset(yOffset + Spatium(2));
```

Comment:
1. `Spatium(2)` is not the correct variable to use, because it returns the score-wide value of spatium. If this staff is scaled, spatium is different. Better to do something like `double yCenteringOffset = 2 * spatium();` (spatium() function always returns the local spatium).
2. The distance between staff lines can be changed too, so it's not _always_ equal to the spatium, so even better would be `double yCenteringOffset = 2 * staff()->lineDist(tick()) * spatium();`
3. You are applying the 2-space offset on top of the already existing offset. Could this lead to strange results if the user has already applied some manual offset? I think it would be safer in this case to override the user offset and just set offset = 2sp for the single line and offset = 0 for all the other cases.

Reply:
@mike-spa thank You. Couple notes:
> 1. `Spatium(2)` is not the correct variable to use, because it returns the score-wide value of spatium. If this staff is scaled, spatium is different. Better to do something like `double yCenteringOffset = 2 * spatium();` (spatium() function always returns the local spatium).
- I think, here no, as it is in Spatium units, so it is magnitude independent.

> 2. The distance between staff lines can be changed too, so it's not _always_ equal to the spatium, so even better would be `double yCenteringOffset = 2 * staff()->lineDist(tick()) * spatium();`
- great idea! But if so, also line distance change on one line staff should recenter it, otherways something like this would happen:

https://github.com/musescore/MuseScore/assets/1646034/565119df-29d3-487a-afeb-dcad62fab600

So I fixed that too in last commit.


> 3. You are applying the 2-space offset on top of the already existing offset. Could this lead to strange results if the user has already applied some manual offset? I think it would be safer in this case to override the user offset and just set offset = 2sp for the single line and offset = 0 for all the other cases.
- Yes, I was thinking of this too, but in some situations, results may be even worse, I think.

Lets say, I have staff with offset set and want to create staff change to oneline staff . With Your proposal, offset is lost and staff is not centered. Than If I change back to 5 line staff, still wrongly offsetted. see:

https://github.com/musescore/MuseScore/assets/1646034/703f7d08-cac2-4215-9ca4-a6b953f0b345

Oroginally, it behaves, I think better:


https://github.com/musescore/MuseScore/assets/1646034/cd12e10a-4235-4005-bd3a-71d4c0f0d605

But Yes, in other situations, your proposal may be bit better (if previous offset is not meant as "general offset, but just like ad-hoc offset)

Changing to oneline in this situation seems to be more logical. But I am not sure, if change back is:

https://github.com/musescore/MuseScore/assets/1646034/aee34484-1059-4869-aa6e-b62a7954892c

In original, change to one line is worse, but change back may is really back to previous state:

https://github.com/musescore/MuseScore/assets/1646034/21a0368e-1945-4628-ad93-4ea7a2715406

I am not sure, if new one is better, but I aplied Your proposal, so it is done new way. 
I can chage it, of course.

P.S. it seems, selection on offset staff needs improvement too :)

Have a nice day!